### PR TITLE
Add other color comparison criteria to findBestfit

### PR DESCRIPTION
**Pull request by Gasparoken**
Implementation of item 1 of https://github.com/aseprite/aseprite/issues/3394

### CODE REVIEW

File: src/doc/palette.cpp [hunk 1]
```diff
@@ -400,6 +408,117 @@ int Palette::findBestfit2(int r, int g, int b, int a) const
   return bestfit;
 }
 
+// Auxiliary function for rgbToOtherSpace()
+double f(double t)
```
Comment (dacap) [inline src/doc/palette.cpp:412]: Add static keyword:
```suggestion
static double f(double t)
```

File: src/doc/palette.cpp [hunk 2]
```diff
@@ -400,6 +408,117 @@ int Palette::findBestfit2(int r, int g, int b, int a) const
...
+{
+  if (t > 0.00885645171)
+    return pow(t, 0.3333333333333333);
```
Comment (dacap) [inline src/doc/palette.cpp:415]: Use `std::` namespace for standard functions. And include `<cmath>`:
```suggestion
    return std::pow(t, 0.3333333333333333);
```

File: src/doc/palette.cpp [hunk 3]
```diff
@@ -400,6 +408,117 @@ int Palette::findBestfit2(int r, int g, int b, int a) const
...
+  else
+    return (t / 0.12841855 + 0.137931034);
+}
+
+// Auxiliary function for findBestfit3()
+void rgbToOtherSpace(double& r, double& g, double& b, FitCriteria fc)
```
Comment (dacap) [inline src/doc/palette.cpp:421]:
```suggestion
static void rgbToOtherSpace(double& r, double& g, double& b, FitCriteria fc)
```

File: src/doc/palette.cpp [hunk 4]
```diff
@@ -400,6 +408,117 @@ int Palette::findBestfit2(int r, int g, int b, int a) const
...
+    Rl = pow((r + 0.055) / 1.055, 2.4);
+  if (g <= 0.04045)
+    Gl = g / 12.92;
+  else
+    Gl = pow((g + 0.055) / 1.055, 2.4);
+  if (b <= 0.04045)
+    Bl = b / 12.92;
+  else
+    Bl = pow((b + 0.055) / 1.055, 2.4);
+  if (fc == linearizedRGB) {
```
Comment (dacap) [inline src/doc/palette.cpp:442]: I think we can use a switch-case here:
```suggestion
  switch (fc) {
    case FitCriteria::linearizedRGB:
      ...
      break;
  }  
```

File: src/doc/palette.cpp [hunk 5]
```diff
@@ -25,6 +25,14 @@ namespace doc {
 
 using namespace gfx;
 
+enum FitCriteria{
```
Comment (dacap) [inline src/doc/palette.cpp:28]: Prefer a `enum class` for global enums (enums outside classes):
```suggestion
enum class FitCriteria {
```

File: src/doc/palette.h [hunk 6]
```diff
@@ -102,6 +102,7 @@ namespace doc {
     bool findExactMatch(color_t color) const;
     int findBestfit(int r, int g, int b, int a, int mask_index) const;
     int findBestfit2(int r, int g, int b, int a) const;
+    int findBestfit3(int r, int g, int b, int a, int mask_index) const;
```
Comment (dacap) [inline src/doc/palette.h:105]: Do we need findBestfit()? Or can we just change the implementation of findBestfit() directly with the new findBest3 code?

**Review by dacap — CHANGES_REQUESTED** [review]

**Comment by Gasparoken** [comment]
Changes just pushed.

**Comment by dacap** [comment]
Cherry-picked https://github.com/aseprite/aseprite/commit/54443ad20d0f0c018704d5f201ca34d39443d61e

**Pull request closed without merging**